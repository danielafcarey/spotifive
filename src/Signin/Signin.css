.signin {
  width: 45%;
  margin: 50px auto;
  text-align: center; }
