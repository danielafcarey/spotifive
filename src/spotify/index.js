const spotifyClientId = '035fb661a97a4ffb938eebc01c0dd4d3';
const spotifyClientSecret = 'bc9e18ee528d4dfa8b9417a6cf5fecf4';
