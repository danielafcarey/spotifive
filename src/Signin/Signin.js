import React from 'react';
import { authorizeSpotify } from '../spotify/authorizeSpotify';

const Signin = (props) => {

  return (
    <div className='signin'>
      <p>
        Spotifive helps you discover new music by generating a playlist of new music to try based on the artists you choose.
      </p>
      <p>To begin, sign in with Spotify:</p>
      <button
        onClick={ handleClick } 
      >Sign in with Spotify</button>
    </div>
  )
}

const handleClick = (event) => {
  authorizeSpotify();
}

export default Signin;
