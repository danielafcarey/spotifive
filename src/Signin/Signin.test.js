import React from 'react';
import { shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import Signin from './Signin';
import * as authorize from '../spotify/authorizeSpotify';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Signin />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('Signin', () => {

  it('matches the snapshot', () => {
    const wrapper = shallow(<Signin />);

    expect(wrapper).toMatchSnapshot();
  })

  describe('handleClick', () => {

    it('calls authorizeSpotify', () => {
      const wrapper = shallow(<Signin />);
      authorize.authorizeSpotify = jest.fn();

      wrapper.find('button').simulate('click');

      expect(authorize.authorizeSpotify).toHaveBeenCalled();
    })
  })

})
