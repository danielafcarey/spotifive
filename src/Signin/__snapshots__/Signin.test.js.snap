// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signin matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Signin />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          Spotifive helps you discover new music by generating a playlist of new music to try based on the artists you choose.
</p>,
        <p>
          To begin, sign in with Spotify:
</p>,
        <button
          onClick={[Function]}
>
          Sign in with Spotify
</button>,
      ],
      "className": "signin",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Spotifive helps you discover new music by generating a playlist of new music to try based on the artists you choose.",
        },
        "ref": null,
        "rendered": "Spotifive helps you discover new music by generating a playlist of new music to try based on the artists you choose.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "To begin, sign in with Spotify:",
        },
        "ref": null,
        "rendered": "To begin, sign in with Spotify:",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign in with Spotify",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Sign in with Spotify",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            Spotifive helps you discover new music by generating a playlist of new music to try based on the artists you choose.
</p>,
          <p>
            To begin, sign in with Spotify:
</p>,
          <button
            onClick={[Function]}
>
            Sign in with Spotify
</button>,
        ],
        "className": "signin",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Spotifive helps you discover new music by generating a playlist of new music to try based on the artists you choose.",
          },
          "ref": null,
          "rendered": "Spotifive helps you discover new music by generating a playlist of new music to try based on the artists you choose.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "To begin, sign in with Spotify:",
          },
          "ref": null,
          "rendered": "To begin, sign in with Spotify:",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign in with Spotify",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Sign in with Spotify",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
